lib/Expression_gen.ml

```diff
@@ -0,0 +1,264 @@
+open Core_kernel
+open Mir
+
+let ends_with suffix s =
+  String.is_suffix ~suffix:suffix s
+
+let starts_with prefix s =
+  String.is_prefix ~prefix:prefix s
+
+let functions_requiring_namespace = 
```

Comment:
I've been using Core's [Set.Poly](https://ocaml.janestreet.com/ocaml-core/108.08.00-pre0//doc/core/Set.Poly.html) package for Sets at the suggestion of Sean and Matthijs. The 'Poly' refers to the fact that the package isn't specialized to holding elements of any particular type, and it'll use OCaml's default 'compare' function to check for equality of elements, which should work as expected in the case of strings. The Set.Poly package inherits the functions of the parent [Set](https://ocaml.janestreet.com/ocaml-core/109.22.00//doc/core/Set.html) package, and most of the documentation for useful functions is there, where `'a` in the type signatures is the type of an element and `('a, 'b) t` is the type of a set of elements.

To build a Set.Poly from this list, you can use the `Set.Poly.of_list` function: 
`let functions_requiring_namespace = Set.Poly.of_list [ ...`